### PR TITLE
Check overlap endpoint

### PR DESCRIPTION
**Pull request by pieterlukasse**
Jira Ticket: [PXP-10006](https://ctds-planx.atlassian.net/browse/PXP-10006)

### New Features
- new endpoint that returns whether two cohorts overlap (after all filtering is done)

### Improvements
- made concept_id int64 to avoid issues w/ big concept_ids. 
- improved exception handling
- fixed test data for HARE concepts

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (UchicagoZchen138) [inline .vscode/launch.json:1]: i think we need to remove this file?

File: utils/parsing.go [hunk 2]
```diff
@@ -19,3 +21,87 @@ func ParseNumericArg(c *gin.Context, paramName string) (int, error) {
...
+	log.Printf("Querying %s: ", paramName)
+	if stringArgValue == "" {
+		log.Printf("bad request - %s should be set", paramName)
+		return "", fmt.Errorf("bad request - %s should set", paramName)
+	} else {
+		return stringArgValue, nil
+	}
+}
+
+func Pos(value int64, list []int64) int {
```
Comment (UchicagoZchen138) [inline utils/parsing.go:49]: always surprised that golang likes to really keep it simple sometimes, like too simple lol

File: controllers/cohortdata.go [hunk 3]
```diff
@@ -156,11 +157,29 @@ func populateConceptValue(row []string, cohortItem models.PersonConceptAndValue,
...
+	var conceptIds []int64
+	sourceId, conceptIds, errors[0] = utils.ParseSourceIdAndConceptIds(c)
+	filterConceptId, errors[1] = utils.ParseBigNumericArg(c, "filterconceptid")
+	filterConceptValue, errors[2] = utils.ParseStringArg(c, "filtervalue")
+	caseCohortId, errors[3] = utils.ParseNumericArg(c, "casecohortid")
+	controlCohortId, errors[4] = utils.ParseNumericArg(c, "controlcohortid")
+	if utils.ContainsNonNil(errors) {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "bad request"})
+		c.Abort()
+	} else {
```
[the remark this answers is not in the document]
Comment (pieterlukasse) [reply controllers/cohortdata.go:174]: good point. A return also works. Let me change that

File: utils/parsing.go (hunk 2 above)
Comment (pieterlukasse) [reply utils/parsing.go:49]: there must be something...right? This is too silly

File: .vscode/launch.json (hunk 1 above)
Comment (pieterlukasse) [reply .vscode/launch.json:1]: I'll remove it.

**Review by UchicagoZchen138 — APPROVED** [review]

**Pull request merged**